### ISSUE TITLE
Make it easier for users to expand the number of protocols they use

### ISSUE THREAD

**Issue by ptgott** [gravitational/teleport#19007]
## Applies To

We'll need to figure this out. Right now, guides to adding resources are in sections called `docs/pages/[a-z]+-access/`, but this will affect the information architectural "glue" between these sections.

## Details

Teleport users often start using Teleport with one protocol in mind, then expand outward to more protocols. Let's improve the information architecture of the docs to make this process easier.

Right now, each protocol has its own top-level section of the docs, e.g., a docs section for "Application Access" and "Server Access". Each section is self contained, with a getting started guide, how-to guides, an access controls reference, and a configuration/CLI reference.

Let's see if we can do more work to choreograph a user's journey _between_ protocol-specific sections to help them:

1. Plan out the protocols they'll add support for
2. Start using those protocols in an integrated way

One effort that could help with this is making our docs UX more like reading a book (https://github.com/gravitational/docs/issues/93). UI components would encourage working linearly through the docs in addition to targeting a single docs page, reading it, then targeting another.

Another thing we could do is add a guide somewhere for planning the protocols a user will add and how they'll go about adding those protocols.

## Related Issues
- Re: the information architecture of our resource guides: https://github.com/gravitational/teleport/issues/13362
- Making the docs UX experience more like a book: https://github.com/gravitational/docs/issues/93

**Comment by ptgott** [comment]
It's worth noting here that users tend to get started with Teleport with the intention of adopting a few protocols, so we should keep the structure of having self-contained, protocol-specific sections.

We can think about how to introduce our protocols for first-time Teleport users so they can plan to expand their usage. Maybe our early-stage docs can introduce our protocols and how to get started with them? E.g., we could include a guide in "Try out Teleport" or even our home section.

**Comment by ptgott** [comment]
Related: https://github.com/gravitational/teleport/issues/22540

**Comment by ptgott** [comment]
Closing this since we now have a docs section dedicated to [general Teleport agent topics](https://github.com/gravitational/teleport/tree/d6ffc0cc389e8013d30b5bbd096e5ae64f5875c4/docs/pages/agents), which we didn't have when I opened this issue. 

There's still more we can do to improve the experience for users who might be interested in using more Teleport protocols, but we should open issues for specific solutions that we can close when those solutions are implemented. That way, we don't need to keep this issue open forever.
